common/txmgr/types/head.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+import "github.com/ethereum/go-ethereum/common"
+
+// Head provides access to a chain's head, as needed by the TxManager.
+// This is a generic interface which ALL chains will implement.
+//
+//go:generate mockery --quiet --name Head --output ./mocks/ --case=underscore
+type Head interface {
+	// BlockNumber is the head's block number
+	BlockNumber() int64
```

Comment:
Is `int64` large enough for all chains?

Reply:
Can't answer that right now. For EVM, it seems large enough, as it is already hard-coded in our codebase. Thus I didn't change it.

For future families, if they require uint256, we will have to convert it to that, or change this to a big.Int kind of interface.

Reply:
probably good enough for now - max int64 (assuming the block generation rate of solana 0.4s/block) would last 116e9 years